### PR TITLE
(#1810869) resolved: Recover missing PrivateTmp=yes and ProtectSystem=struct

### PR DESCRIPTION
**Pull request by d-hatayama**
Since the commit b61e8046ebcb28225423fc0073183d68d4c577c4,
systemd-resolved.service often fails to start with the following message:

    Failed at step NAMESPACE spawning /usr/bin/mount: Read-only file system

This is because dropping DynamicUser=yes dropped implicit PrivateTmp=yes and
also implicit After=systemd-tmpfiles-setup.service, and thus
systemd-resolved.service can start before systemd-remount-fs.service. As a
result, mount operations associated with PrivateDevices= can be performed to
still read-only filesystems.

To fix this issue, it's better to recover PrivateTmp=yes and
ProtectSystem=struct just as the upstream commit
62fb7e80fcc45a1530ed58a84980be8cfafa9b3e (Revert "resolve: enable DynamicUser=
for systemd-resolved.service").

### CODE REVIEW

**Comment by lnykryn** [comment]
@d-hatayama Can I ask you to also file an rh bugzilla and add
Resolves: #BZ_NUM
to the commit message? It is kinda needed to get it through our automation.

**Comment by d-hatayama** [comment]
I see. Please wait for a while.

**Comment by d-hatayama** [comment]
I created BZ#1810869 and added Resolves: #1810869.

I also corrected two typo in the commit description: "struct" should have been "strict".

**Review by jsynacek — APPROVED** [review]
LGTM.

**Review by jsynacek — APPROVED** [review]
LGTM.

File: units/systemd-resolved.service.in [hunk 1]
```diff
@@ -28,7 +28,9 @@ WatchdogSec=3min
 User=systemd-resolve
 CapabilityBoundingSet=CAP_SETPCAP CAP_NET_RAW CAP_NET_BIND_SERVICE
 AmbientCapabilities=CAP_SETPCAP CAP_NET_RAW CAP_NET_BIND_SERVICE
+PrivateTmp=yes
 PrivateDevices=yes
+ProtectSystems=strict
```
Comment (olifre) [inline units/systemd-resolved.service.in:33]: Shouldn't this be:
```
ProtectSystem=strict
```
i.e. without the trailing `s`? 
Using it as-is, I get:
```
Unknown lvalue 'ProtectSystems' in section 'Service'
```

**Pull request merged**